src/url-params.ts

```diff
@@ -175,12 +175,18 @@ export class AnalyticsUrlParams {
 	/**
 	 * @return full query parameter string that can be appended to URLs
 	 */
-	getQueryString(destinationUrl?: URL, currentUrl?: URL): string {
+	getQueryString(destinationUrl?: URL | string, currentUrl?: URL): string {
+		// we first check if destionationUrl is a relative URL string. If it is, we exit.
+		//const relativeRegex = /^(?!www\.|(?:http|ftp)s?:\/\/|[A-Za-z]:\\|\/\/).*/g;
+		let destinationDomainMatch;
+		let destinationDomain;
+		let actualDomainMatch;
+		let actualDomain;
+
 		// this regex is based on the assumption that we wont be using TLDs longer than 3 characters. If we do, it will break
 		// the logic and take that longer TLD as the main domain, for example hub.balena.edge.io -> edge.io
 		const regex = /([a-zA-Z0-9-]+)(\.[a-zA-Z]{2,3})?(\.[a-zA-Z]+$)/g;
 
-		let actualDomainMatch;
```

Comment:
I actually prefer it here, next to where it's first assigned, so that's easier to follow.
Also type it as:
```
let actualDomainMatch: RegExpMatchArray | null;
```